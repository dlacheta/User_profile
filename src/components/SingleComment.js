import React from 'react';
import PropTypes from 'prop-types';
import dateDiffInDays from '../func/dateDiffInDays';

const SingleComment = ({
  comment: {
    name, comment, date, avatar,
  },
}) => (
  <div className="comments-scroll">
    <img className="comment-img" src={avatar} alt={name} />
    <div className="container-fluid">
      <div className="comment-header">
        <h4>{name}</h4>
        <span className="comment-date">{dateDiffInDays(date)}</span>
      </div>
      {comment}
    </div>
  </div>
);

SingleComment.propTypes = {
  comment: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    date: PropTypes.number,
    comment: PropTypes.string,
  }),
};

SingleComment.defaultProps = {
  comment: {},
};

export default SingleComment;
