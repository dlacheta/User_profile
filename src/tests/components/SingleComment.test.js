import React from 'react';
import { shallow } from 'enzyme';
import SingleComment from '../../components/SingleComment';

it('SingleComment component renders correctly', () => {
  const wrapper = shallow(<SingleComment />);
  expect(wrapper).toMatchSnapshot();
});
