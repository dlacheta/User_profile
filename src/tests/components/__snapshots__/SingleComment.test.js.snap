// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleComment component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleComment
    comment={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          className="comment-img"
        />,
        <div
          className="container-fluid"
        >
          <div
            className="comment-header"
          >
            <h4 />
            <span
              className="comment-date"
            >
              NaNd
            </span>
          </div>
        </div>,
      ],
      "className": "comments-scroll",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "className": "comment-img",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="comment-header"
            >
              <h4 />
              <span
                className="comment-date"
              >
                NaNd
              </span>
            </div>,
            undefined,
          ],
          "className": "container-fluid",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4 />,
                <span
                  className="comment-date"
                >
                  NaNd
                </span>,
              ],
              "className": "comment-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "NaNd",
                  "className": "comment-date",
                },
                "ref": null,
                "rendered": "NaNd",
                "type": "span",
              },
            ],
            "type": "div",
          },
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="comment-img"
          />,
          <div
            className="container-fluid"
          >
            <div
              className="comment-header"
            >
              <h4 />
              <span
                className="comment-date"
              >
                NaNd
              </span>
            </div>
          </div>,
        ],
        "className": "comments-scroll",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "className": "comment-img",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="comment-header"
              >
                <h4 />
                <span
                  className="comment-date"
                >
                  NaNd
                </span>
              </div>,
              undefined,
            ],
            "className": "container-fluid",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4 />,
                  <span
                    className="comment-date"
                  >
                    NaNd
                  </span>,
                ],
                "className": "comment-header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "NaNd",
                    "className": "comment-date",
                  },
                  "ref": null,
                  "rendered": "NaNd",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
